### PR TITLE
feat(schematron): verify string value of Schematron property

### LINKED ISSUE #2044: Lack of support for testing text output produced by Schematron property elements
I've noticed that XSpec doesn't support testing the text content of Schematron `properties` in the output produced by an `assert` or `report` (unlike `diagnostics` output).

The [Writing Scenarios for Schematron](https://github.com/xspec/xspec/wiki/Writing-Scenarios-for-Schematron) Wiki page has the following, which looks like `property` output is specifically excluded: "Starting in XSpec v3.0.3, the `x:expect-assert` and `x:expect-report` elements can also have text content that identifies an expected `assert` or `report` by its message or diagnostic."

Is there a reason for this?

### PR DESCRIPTION
**Pull request by galtm**
Support verification of the space-normalized string values of Schematron properties, using text content of the `x:expect-assert` and `x:expect-report` elements. This enhancement extends #1822.

Closes #2044.

### Notes

1. The skeleton implementation of Schematron produces different SVRL markup for property references, and this pull request does _not_ attempt to support the skeleton implementation. **Reasons:** (A) It would clutter both the XSpec compiler and the compiled test code (potentially slowing it down because the processor would have more markup patterns to search for); (B) The [XSpec wiki](https://github.com/xspec/xspec/wiki/Using-Another-Implementation-of-Schematron#using-skeleton-implementation-with-xspec-v30-not-supported) says that XSpec no longer supports the skeleton implementation of Schematron starting in XSpec v3.0.
2. In the code change in `src/schematron/schut-to-xspec.xsl`, the left side of the '=' expression could instead say `(descendant::{x:known-UQName('svrl:text')}) ! normalize-space(.)` but https://github.com/Schematron/schematron-enhancement-proposals/issues/79 raises SVRL markup design questions. To be defensive against future SVRL changes in Schematron, it's safer to list the child elements of interest.
3. Without this change, only a main message or diagnostic can satisfy the constraint imposed by the text content inside `x:expect-assert` or `x:expect-report`. Theoretically, a user might prefer the v3.0/v3.1 behavior, i.e., verify the string value of a main message or diagnostic and ignore a property that just happens to have the same string value. It seems so unlikely that I'm not inclined to call it out in the release notes as an "incompatibility."

### CODE REVIEW

**Comment by galtm** [comment]
On the GitHub Actions test page, I see warnings that say:
```
ubuntu-latest pipelines will use ubuntu-24.04 soon. For more details, see https://github.com/actions/runner-images/issues/10636
```

When this commit ran in my own fork, it used ubuntu 24, not 22. (I expanded "Set up job" and then expanded "Operating system".) It seems to suggest that the tests we run will not be negatively affected by the Ubuntu upgrade.

File: test/schematron/schut-to-xspec-012-out.xspec [hunk 1]
```diff
@@ -44,9 +44,9 @@
          label="with text">
          <x:context href="%TEST_BASE%/schematron/schut-to-xspec-012-02.xml"/>
          <x:expect label="assert text: A section inside a div is usually superfluous"
-                   test="exists(Q{http://purl.oclc.org/dsdl/svrl}schematron-output/Q{http://purl.oclc.org/dsdl/svrl}failed-assert[&#xA;            (Q{http://purl.oclc.org/dsdl/svrl}text, Q{http://purl.oclc.org/dsdl/svrl}diagnostic-reference) ! normalize-space(.)&#xA;            = 'A section inside a div is usually superfluous'&#xA;            ])"/>
+            test="exists(Q{http://purl.oclc.org/dsdl/svrl}schematron-output/Q{http://purl.oclc.org/dsdl/svrl}failed-assert[&#xA;            (&#xA;            Q{http://purl.oclc.org/dsdl/svrl}text,&#xA;            Q{http://purl.oclc.org/dsdl/svrl}diagnostic-reference,&#xA;            Q{http://purl.oclc.org/dsdl/svrl}property-reference&#xA;            ) ! normalize-space(.)&#xA;            = 'A section inside a div is usually superfluous'&#xA;            ])"/>
```
Comment (galtm) [inline test/schematron/schut-to-xspec-012-out.xspec:47]: FYI: This file is used by `test/schematron/schut-to-xspec.xspec`. The changes in this file reflect the changes in `schut-to-xspec.xsl`.

**Review by birdya22 — COMMENTED** [review]
I've had a look at the changes and don't have any comments.

**Comment by galtm** [comment]
> I've had a look at the changes and don't have any comments.

@birdya22 , thank you for looking at them!

**Pull request merged**